File: Sprint02/frontend/src/types/scans.ts
export interface ScanSummary {
  scan_id: string;
  repository_url: string;
  status: string;
  deployment_approved: boolean;
  started_at: string;
  completed_at: string | null;
  critical_count: number;
  high_count: number;
  medium_count: number;
  low_count: number;
  secrets_count: number;
}

export interface Finding {
  scanner_type: string;
  severity: string;
  file_path: string | null;
  line_number: number | null;
  title: string | null;
  description: string | null;
  vulnerability_id: string | null;
}

export interface ScanDetails extends ScanSummary {
  gitleaks_findings: Finding[];
  trivy_findings: Finding[];
}

export interface CreateScanResponse {
  scan_id: string;
  status: string;
  deployment_approved: boolean;
  critical_count: number;
  high_count: number;
  medium_count: number;
  secrets_count: number;
}

File: Sprint02/backend/app/schemas/scans.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ScanCreate(BaseModel):
    repository_url: str = Field(
        ...,
        examples=["https://github.com/org/project"],
        description="Git repository URL to clone and scan.",
    )


class ScanCreateResponse(BaseModel):
    scan_id: str = Field(examples=["SCAN-20260622-0001"])
    status: str = Field(examples=["PASSED"])
    deployment_approved: bool = Field(examples=[True])
    critical_count: int = Field(examples=[0])
    high_count: int = Field(examples=[3])
    medium_count: int = Field(examples=[8])
    secrets_count: int = Field(examples=[0])


class FindingResponse(BaseModel):
    scanner_type: str
    severity: str
    file_path: str | None = None
    line_number: int | None = None
    title: str | None = None
    description: str | None = None
    vulnerability_id: str | None = None

    model_config = ConfigDict(from_attributes=True)


class ScanSummary(BaseModel):
    scan_id: str
    repository_url: str
    status: str
    deployment_approved: bool
    started_at: datetime
    completed_at: datetime | None = None
    critical_count: int
    high_count: int
    medium_count: int
    low_count: int
    secrets_count: int

    model_config = ConfigDict(from_attributes=True)


class ScanDetailsResponse(ScanSummary):
    gitleaks_findings: list[FindingResponse]
    trivy_findings: list[FindingResponse]
